### PR TITLE
tests(ci): authenticate protoc download

### PR DESCRIPTION
**Pull request by brendankenny**
running into github api rate limiting errors when downloading protoc, which is weird because it's github making the request. Turns out we need to supply a token :)

https://github.com/arduino/setup-protoc/issues/6

Also drops CI runs on `push`

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -27,6 +27,7 @@ jobs:
       uses: arduino/setup-protoc@master
```
Comment (paulirish) [inline .github/workflows/ci.yml:27]: should we lock to this version of the action so we dont leak the token to whatever lands in their master tomorrow?

i think `arduino/setup-protoc@v1.1.0` will work? if not we'll need to ask them to rls

**Comment by paulirish** [comment]
Here's where the token ends up being used: https://github.com/arduino/setup-protoc/blob/7ad700d3b20e2a32b35d2c17fbdc463891608381/lib/installer.js#L125-L136

File: .github/workflows/ci.yml (hunk 1 above)
Comment (paulirish) [reply .github/workflows/ci.yml:27]: jk we can do a specific commit: `arduino/setup-protoc@7ad700d`

thx https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsuses

Comment (paulirish) [inline .github/workflows/ci.yml:27]:
```suggestion
      uses: arduino/setup-protoc@7ad700d
```

Comment (patrickhulce) [reply .github/workflows/ci.yml]: oh that's so cool! it never occurred to me that github actions can keep secrets from being in the environment already unlike travis. that's dope :)

**Pull request merged**